video/audio_sample.h

```diff
@@ -9,35 +9,36 @@
 #include "buffer_stream.h"
 
 struct AudioSample {
-	AudioSample(std::vector<std::shared_ptr<BufferStream>>& streams, uint8_t format, uint32_t sampleRate = AUDIO_DEFAULT_SAMPLE_RATE, uint16_t frequency = 0) :
-		blocks(streams), format(format), sampleRate(sampleRate), baseFrequency(frequency) {}
+	AudioSample(std::vector<std::shared_ptr<BufferStream>>& streams, uint16_t id, uint8_t format, uint32_t sampleRate = AUDIO_DEFAULT_SAMPLE_RATE, uint16_t frequency = 0) :
+		blocks(streams), id(id), format(format), sampleRate(sampleRate), baseFrequency(frequency) {}
 	~AudioSample();
 
 	int8_t getSample(uint32_t & index, uint32_t & blockIndex);
 	void seekTo(uint32_t position, uint32_t & index, uint32_t & blockIndex, int32_t & repeatCount);
 	uint32_t getSize();
 
 	std::vector<std::shared_ptr<BufferStream>>& blocks;
+	uint16_t		id;					// ID of the sample
```

Comment:
this ID was added in purely for debugging - it's arguably superfluous and may be got rid of